### DISCUSSION TITLE
Milvus Backup Gets Stuck

### DISCUSSION THREAD

**Discussion by radhamde** [milvus-io/milvus#31195] [Q&A and General discussion]
When we are running milvus-backup from our local system by connecting remote milvus and minio, the backup process stops randomly at any point of the process and holds forever on it. We are facing this sort of message when we are getting stuck 

[2024/03/11 23:12:54.327 +05:30] [INFO] [core/backup_impl_create_backup.go:276] ["field index"] [collection_name=DHLOracleEBS] [field_name=vector] ["index info"="[{}]"]
[2024/03/11 23:12:55.967 +05:30] [INFO] [core/backup_impl_create_backup.go:364] ["GetPersistentSegmentInfo before flush from milvus"] [collectionName=DHLOracleEBS] [segmentNumBeforeFlush=4]

We are using this details of parallelism inside our backup config 

backup:
  maxSegmentGroupSize: 2G

  parallelism:
    # collection level parallelism to backup
    backupCollection: 4
    # thread pool to copy data. reduce it if blocks your storage's network bandwidth
    copydata: 128
    # Collection level parallelism to restore
    restoreCollection: 2

  # keep temporary files during restore, only use to debug
  keepTempFiles: false

Could you please advise about any fix?

Regards
Radha M De

**Comment by yhmo** [comment]
Seems the backup tool is stuck on calling the flush() interface.
Which version of the milvus-backup?
And which version of your milvus server?

**Comment by radhamde** [comment]
Milvus Backup 

./milvus-backup --version
0.4.4 (Built on 2023-12-12T06:34:36Z from Git SHA fca8663a31f95da7f0e6ecd1f945f4b6b7bc540a)
milvus-backup version 0.4.4 (Built on 2023-12-12T06:34:36Z from Git SHA fca8663a31f95da7f0e6ecd1f945f4b6b7bc540a)

Milvus Server Version 2.2.13

Regards
Radha M De

**Reply by yhmo** [reply]
I think the reason is: the rpc proto is changed between v2.2.13 and v2.3.x.
Your milvus server is 2.2.13.
According to [this version-match matrix](https://github.com/zilliztech/milvus-backup?tab=readme-ov-file#compatibility):  the milvus-backup 0.4.4 is compatible with milvus v2.3.x.
I did see the proto changed for flush() interface between v2.2.x and v2.3.x.  I think you should use milvus-backup 0.3.3.

**Comment by radhamde** [comment]
Thanks. Let me plan to bring down 0.3.3 milvus-backup for use.

**Comment by debjotimallick** [comment]
Milvus backup is still getting stuck when using version 0.3.3. It's stuck in the following lines:

`[2024/03/12 19:24:08.769 +05:30] [INFO] [core/backup_impl_create_backup.go:540] ["partition size is smaller than MaxSegmentGroupSize, won't separate segments into groups in backup files"] [collectionId=446448523559131357] [partitionId=446448523559131358] [partitionSize=197738] [MaxSegmentGroupSize=2147483648]
[2024/03/12 19:24:11.973 +05:30] [INFO] [core/backup_impl_create_backup.go:398] ["GetPersistentSegmentInfo before flush from milvus"] [collectionName=testCollection] [segmentNumBeforeFlush=3]`

**Reply by yhmo** [reply]
Could you provide the logs of the milvus server?

**Comment by radhamde** [comment]
Hello yhmo
Our setup for Milvus is as openshift container-based and TLS-enabled. Since the Milvus-backup server cannot work with TLS, we disable the TLS and restart the container to take Milvus backup. Once the backup is done, we enable TLS in Milvus and restart the container to enable a secured Milvus container. 

So, the milvus server logs may not be helpful to find details of the problem which happened during milvus backup.

Any thoughts please?

**Reply by yhmo** [reply]
Use pymilvus v2.2.17 to call collection.flush() manually. If the call is stuck, you can use **docker logs [container_id] > 1.log** to export the log, and send the "1.log" to me.

**Comment by debjotimallick** [comment]
Hi yhmo,

In the version 0.4.4 of milvus-backup server, we see that **--force** flag has been introduced, which says "force backup skip flush, should make sure data has been stored into disk when using it". As we were getting stuck in "GetPersistentSegmentInfo before flush from milvus", can using this flag skip the flush step and complete the backup inside minio? Are there any adverse effects of using this flag?

Thanks,
Debjoti Mallick

**Reply by yhmo** [reply]
Yes, if "force" is true, it will skip flush(). My concern is why it is stuck at flush, and some data might not be able to backup without flush.

**Reply by yhmo** [reply]
You can use pymilvus v2.2.17 to call collection.flush() manually, to observe if it is stuck, too.

**Reply by SaubhikBanerjee** [reply]
Yes, manual collection.flush() is getting stuck too. What can be the reason?

**Reply by yhmo** [reply]
flush stuck could be a bug. We need the log to investigate. If your milvus is deployed by docker, you can use **docker logs [container_id] > 1.log** to export the log, and send the "1.log" to me.
